Update Account to namespace, fix docs (#297)
OpenZeppelin/cairo-contracts: docs/Account.md

This is the basic flow:

1. The user sends the messages for the transaction through a Signer instantiation which looks like this:

    ```python
    await signer.send_transaction(
            account, [
                (contract_address, 'contract_method', [arg_1]),
                (contract_address, 'another_method', [arg_1, arg_2])
            ]
        )
    ```

    The `_from_call_to_call_array` method in [utils.py](../tests/utils.py) converts each call into the `AccountCallArray` format and cumulatively stores the calldata of every call into a single array. Next, both arrays (as well as the `sender`, `nonce`, and `max_fee`) are used to create the transaction hash. The Signer then invokes `__execute__` with the signature and passes `AccountCallArray`, calldata, and nonce as arguments.

> 2. The `__execute__` method creates an empty array which is passed with the `AccountCallArray` and calldata into the `_from_call_array_to_call` method. This method iterates through the `AccountCallArray` messages and converts each message into a `Call`.

Reviewer: ```suggestion
2. The `__execute__` method takes the `AccountCallArray` and calldata and builds an array of `Call` (MultiCall).
```
I think this is simpler

> 3. A new empty array is created and passed with the list of calls (populated from `from_call_array_to_call`) to `_execute_list`. This method iterates and executes a contract call for each `Call` and pushes each result into the array. Once all contract calls are finished, the array holding each contract call response is returned.

Reviewer: I think this is just an implementation detail, it's the interface what we want to document, right?

Author: Correct. I broke this down for myself so I could better explain/understand the process. A blog post or something might be better suited to document the flow. Good call

Reviewer: I think it's a bit confusing to go that deep into this kind details, e.g. we could say "An array of `Call`s is passed to `_execute_list`" instead -- no need to explain that a new array is allocated, iterated, etc. I'd remove this section alltogether.
